go-memory-allocation/README_zh-CN.md

```
@@ -0,0 +1,687 @@
+
+在本文中，我们将深入探讨 Go 的内存分配器。我们会查看其核心组件、如何协同为不同大小的分配服务，以及栈如何与堆对象一起管理。并穿插一些案例研究，让你理解 Go 内存分配策略的实际影响。读完后，你应能更清晰地看到 Go 是如何在屏蔽内存管理细节的同时仍保持高性能的。
+
+在深入了解 Go 如何分配内存之前，先理解典型操作系统中内存工作的基础概念是很重要的。我建议先读一遍 [虚拟内存基础（Fundamental of Virtual Memory）](https://nghiant3223.github.io/2025/05/29/fundamental_of_virtual_memory.html) 那篇文章。如果你已经熟悉相关概念，可以跳过那篇。现在我们进入 Go 对虚拟内存的视角。
+
+## Go 对虚拟内存的视角
+
+由于一个 Go 进程本质上是用户态应用程序，它遵循在那篇基础文章中描述的标准虚拟内存布局。特别地，进程的 _Stack_ 段是与 Go 运行时主线程 `m0` 关联的 `g0` 栈（即系统栈）。已初始化的全局变量存放在 _Data_ 段，而未初始化的则位于 _BSS_ 段。
+
+传统意义上的 _Heap_ 段（位于程序断点 program break 之下）并不是 Go 运行时用于分配堆对象的位置。相反，Go 运行时大量依赖**内存映射段（memory-mapped segments）**来为堆对象与 goroutine（协程）栈分配内存。今后我把 Go 用于动态分配的这些内存映射段称为“堆”，这与传统的程序断点下的堆不要混淆。
+
+| <img src="go_virtual_memory_view.png" width=300> |
```

markdownlint-cli2 (0.18.1)

Images should have alternate text (alt text)

(MD045, no-alt-text)